website/docs: Password Uniqueness Policy (#13686)
goauthentik/authentik: website/docs/customize/policies/unique_password.md

:::

## Integration with other policies

For comprehensive password security, consider using this policy alongside:

- [Password Policy](./index.md#password-policy) - To enforce password complexity rules
- [Password-Expiry Policy](./index.md#password-expiry-policy) - To enforce regular password rotation

## Implement a Password Uniqueness policy

To implement a policy that prevents users from reusing their previous passwords, follow these steps:

1. In the Admin interface, navigate to **Customization** > **Policies**.
2. Click **Create** to define a new Password Uniqueness Policy.
    - **Name**: provide a descriptive name for the policy.
    - **Password field**: enter the name of the input field to check for the new password. By default, if no custom flows are used, the field name is `password`. This field name must match the field name used in your Prompt stage.
    - **Number of previous passwords to check**: enter the number of past passwords that you want to set as the number of previous passwords that are checked and stored for each user, with a default of 1. For instance, if set to 3, users will not be able to reuse any of their last 3 passwords.

> 3. Bind the policy to your **password prompt stage**: For example, if you're using the `default-password-change` flow, edit the `default-password-change-prompt` stage and add the policy in the Validation Policies section.

Reviewer: ```suggestion
3. Bind the policy to your **password prompt stage**: For example, if you're using the `default-password-change` flow, edit the `default-password-change-prompt` stage and add the policy in the **Validation Policies** section.
```

---
title: Password Uniqueness Policy
sidebar_label: Password Uniqueness Policy
support_level: authentik
tags:
    - policy
    - password
    - security
    - enterprise
authentik_version: "2025.4.0"
authentik_enterprise: true
---

The Password Uniqueness Policy prevents users from reusing their previous passwords when setting a new password.

## How it works

> This policy stores a history of user passwords and checks any new password against this history. If a match is found with a previous password, the policy fails and the user is required to choose a different password.

Reviewer 2: ```suggestion
This policy maintains a record of previously used passwords for each user. When a new password is created, it is compared against this historical log. If a match is found with any previous password, the policy is not met, and the user is required to choose a different password.
```

What do you think of this wording?

The password history is maintained automatically when this policy is in use. Old password hashes are stored securely in authentik's database.

## Configuration options

The Password Uniqueness Policy has two configuration options:

- **Password Field**: The field key to check for the new password. Default is _password_. This should match the field name used in your Prompt stage.

- **Number of Historical Passwords**: The number of previous passwords to check against. Default is 1. This also controls how many old passwords the system stores for each user.

>     For example, setting this to 3 ensures a user cannot reuse any of their last 3 passwords.

Reviewer 2: Is there a reason for indenting this line? I assume "this" refers to the number of historical passwords? We could possibly merge this line with the text on line 28? Here's what I would do:

```
This setting determines how many previous passwords are checked and stored for each user, with a default of 1. For instance, if set to 3, users will not be able to reuse any of their last 3 passwords.

## Integration with other policies

For comprehensive password security, consider using this policy alongside:

- [Password Policy](./index.md#password-policy) - To enforce password complexity rules
- [Password-Expiry Policy](./index.md#password-expiry-policy) - To enforce regular password rotation

## Implementation example

To implement a policy that prevents users from reusing their last 3 passwords:

> 1. Create a **Password Uniqueness Policy**

Reviewer 2: Can we add a little introduction telling the user how to reach this page?

---
title: Password Uniqueness Policy
sidebar_label: Password Uniqueness Policy
support_level: authentik
tags:
    - policy
    - password
    - security
    - enterprise
authentik_version: "2025.4.0"
authentik_enterprise: true
---

The Password Uniqueness Policy prevents users from reusing their previous passwords when setting a new password.

## How it works

This policy maintains a record of previously used passwords for each user. When a new password is created, it is compared against this historical log. If a match is found with any previous password, the policy is not met, and the user is required to choose a different password.

The password history is maintained automatically when this policy is in use. Old password hashes are stored securely in authentik's database.

:::info
This policy takes effect after the first password change following policy activation. Before that first change, there's no password history data to compare against.
:::

> ## Configuration options

Reviewer: @melizeche I *think we can actually remove this section, because both of those fields should be fully described down in your procedural steps, where they are creating the policy. If you want to leave this section, to highlight those two config settings, we can but let's pare it down to describe the fields but no go into great detail and duplicate what is in the steps. Thoughts?

The Password Uniqueness Policy has two configuration options:

> - **Password Field**: The field key to check for the new password. Default is _password_. This should match the field name used in your Prompt stage.

Reviewer 2: ```suggestion
- **Password Field**: The field key to check for the new password. Default is `password`. This should match the field name used in your Prompt stage.
```

> - **Password Field**: The field key to check for the new password. Default is `password`. This should match the field name used in your Prompt stage.

Reviewer: ```suggestion
- **Password Field**: Enter the name of the input field to check for the new password. By default, if no custom flows are used, the field name is `password`. This field name should match the field name used in your Prompt stage.
```

Reviewer: @marcelo, ideally we can explain a bit more about how the Prompt stage is involved. And is it a "should" or a "must" math?

- **Number of Historical Passwords**: This setting determines how many previous passwords are checked and stored for each user, with a default of 1. For instance, if set to 3, users will not be able to reuse any of their last 3 passwords.

## Integration with other policies

For comprehensive password security, consider using this policy alongside:

- [Password Policy](./index.md#password-policy) - To enforce password complexity rules
- [Password-Expiry Policy](./index.md#password-expiry-policy) - To enforce regular password rotation

> ## Implementation example

Reviewer: ```suggestion
## Implement a Password Uniqueness Policy
```

> To implement a policy that prevents users from reusing their last 3 passwords:

Reviewer: ```suggestion
To implement a policy that prevents users from reusing their previous passwords, follow these steps:
```

> 1. In the Admin interface, navigate to **Customization** -> **Policies**.

Reviewer: ```suggestion
1. In the Admin interface, navigate to **Customization** > **Policies**.
```

> 2. Create a **Password Uniqueness Policy**

Reviewer: ```suggestion
2. Click **Create** to define a new Password Uniqueness Policy.
```

3. Set **Number of Historical Passwords** to 3

> 4. Bind the policy to your **password prompt stage**

Reviewer: @melizeche is the *password prompt stage* either a custom one that created or also the `default-password-change-prompt ` stage?

Reviewer: Let's give them a bit more help on how to do the binding.
